### ISSUE TITLE
Incorrect error when constructing a dataclass with a dataclass.InitVar

### ISSUE THREAD

**Issue by jfly** [microsoft/pyright#2493]
I have a very simple dataclass with an `InitVar`, and pyright is telling me I cannot pass a string as a constructor parameter, when I definitely should be allowed to.

**To Reproduce**

In a fresh Python 3.6.9 virtualenv:

```
$ python --version
Python 3.6.9
$ pyright --version
pyright 1.1.181
$ pip install dataclasses
$ echo 'import dataclasses

@dataclasses.dataclass
class Foo:
    bar: dataclasses.InitVar[str]

Foo(bar="this is a string")' > demo.py
$ echo '{}' > pyrightconfig.json
$ pyright demo.py
Loading configuration file at /home/jeremy/tmp/dcing/pyrightconfig.json
Assuming Python version 3.6
Assuming Python platform Linux
Auto-excluding **/node_modules
Auto-excluding **/__pycache__
Auto-excluding .git
stubPath /home/jeremy/tmp/dcing/typings is not a valid directory.
Searching for source files
Found 1 source file
/home/jeremy/tmp/dcing/demo.py
  /home/jeremy/tmp/dcing/demo.py:9:9 - error: Argument of type "Literal['this is a string']" cannot be assigned to parameter "bar" of type "InitVar[str]" in function "__init__"
    "Literal['this is a string']" is incompatible with "InitVar[str]" (reportGeneralTypeIssues)
1 error, 0 warnings, 0 infos
Completed in 0.593sec
```

**Expected behavior**
Pyright shouldn't error out in this situation, I should be allowed to pass a string in without error.

**Additional context**
No idea if this is useful to know, but if I remove the `pyrightconfig.json` file, the problem goes away:

```
$ rm pyrightconfig.json
$ pyright demo.py
 No configuration file found.
No pyproject.toml file found.
stubPath /home/jeremy/tmp/dcing/typings is not a valid directory.
Assuming Python platform Linux
Searching for source files
Found 1 source file
0 errors, 0 warnings, 0 infos
Completed in 0.602sec
```

Also, I am unable to reproduce this problem in a Python 3.8 virtualenv. So maybe this is related to the [`dataclasses` backport for Python 3.6](https://pypi.org/project/dataclasses/)?
